agent-shell.el

```diff
@@ -1429,6 +1430,15 @@ Must provide ON-SESSION-INIT (lambda ())."
    :on-success (lambda (response)
                  (map-put! agent-shell--state
                            :session-id (map-elt response 'sessionId))
+                 ;; Store available modes and current mode
+                 (when-let ((modes-data (map-elt response 'modes)))
+                   (setf (map-elt agent-shell--state :available-modes)
+                         (map-elt modes-data 'availableModes))
+                   (setf (map-elt agent-shell--state :current-mode-id)
+                         (map-elt modes-data 'currentModeId))
+                   ;; Update modeline to show mode
+                   (with-current-buffer (map-elt agent-shell--state :buffer)
```

Comment:
with-current-buffer should't be needed anymore after https://github.com/xenodium/agent-shell/commit/3936be586dbaf9772f9104551f172f7b948a98cb Did you run into issues?

acp should now invoke callbacks with the intended context buffer